### PR TITLE
Action Logs - hidden secret

### PR DESCRIPTION
**Pull request by brianteeman**
If you want to search in the list of actions to find a specific action then there is more that you can do that simply search in the title. But there is no way to know this without reading the code.

This PR adds the full information for searching  as a tooltip. This is in exactly the same format as in other components.
> COM_ACTIONLOGS_FILTER_SEARCH_DESC="Search in username. Prefix with ID: to search for an actionlog ID. Prefix with ITEM_ID: to search for an actionlog item_id. Prefix with UID: to search for a user ID."

As it was easy to do and could be considered a bug fix I **added** the ability to search for a user_id

### CODE REVIEW

**Comment by alikon** [comment]
humm, we can already search for a user 
Screenshot from 2019-10-21 19-39-51

maybe i'm reading this too fast ...

**Comment by Quy** [comment]
PR is for J3.

File: administrator/language/en-GB/en-GB.com_actionlogs.ini [hunk 1]
```diff
@@ -24,6 +24,7 @@ COM_ACTIONLOGS_EXTENSION_ASC="Extension ascending"
 COM_ACTIONLOGS_EXTENSION_DESC="Extension descending"
 COM_ACTIONLOGS_EXTENSION_FILTER_DESC="Search in the User Action logs by extension"
 COM_ACTIONLOGS_ERROR_COULD_NOT_EXPORT_DATA="Could not export data."
+COM_ACTIONLOGS_FILTER_SEARCH_DESC="Search in username. Prefix with ID: to search for an actionlog ID. Prefix with ITEM_ID: to search for an actionlog item_id. Prefix with UID: to search for a user ID."
```
Comment (Quy) [inline administrator/language/en-GB/en-GB.com_actionlogs.ini]:
```suggestion
COM_ACTIONLOGS_FILTER_SEARCH_DESC="Search in username. Prefix with ID: to search for an actionlog ID. Prefix with ITEM_ID: to search for an actionlog item ID. Prefix with UID: to search for a user ID."
```

**Comment by alikon** [comment]
yes .... but iirc the same feature is present in j3....  if not forgive me in advance

**Comment by brianteeman** [comment]
@alikon It does have that filter for a user by username. This added the ability to search by the user_id

File: administrator/language/en-GB/en-GB.com_actionlogs.ini (hunk 1 above)
Comment (brianteeman) [reply administrator/language/en-GB/en-GB.com_actionlogs.ini]: i kept it as item_id as I have zero idea what the field contains ;)

**Comment by alikon** [comment]
i was quite sure i'ved missing something....
thanks for the  clarification

File: administrator/language/en-GB/en-GB.com_actionlogs.ini (hunk 1 above)
Comment (Quy) [reply administrator/language/en-GB/en-GB.com_actionlogs.ini]: It is the id of a tag, module, banner, etc that was performed to.

File: administrator/language/en-GB/en-GB.com_actionlogs.ini [hunk 2]
```diff
@@ -24,6 +24,7 @@ COM_ACTIONLOGS_EXTENSION_ASC="Extension ascending"
 COM_ACTIONLOGS_EXTENSION_DESC="Extension descending"
 COM_ACTIONLOGS_EXTENSION_FILTER_DESC="Search in the User Action logs by extension"
 COM_ACTIONLOGS_ERROR_COULD_NOT_EXPORT_DATA="Could not export data."
+COM_ACTIONLOGS_FILTER_SEARCH_DESC="Search in username. Prefix with ID: to search for an actionlog ID. Prefix with ITEM_ID: to search for an actionlog item ID. Prefix with UID: to search for a user ID."
```
Comment (Quy) [inline administrator/language/en-GB/en-GB.com_actionlogs.ini]:
```suggestion
COM_ACTIONLOGS_FILTER_SEARCH_DESC="Search in username. Prefix with ID: or ITEM_ID: or UID: to search for an actionlog ID, an item ID or a user ID."
```

Comment (Quy) [reply administrator/language/en-GB/en-GB.com_actionlogs.ini]: To be consistent with Articles search filter and shorter.

> Search in title, alias and note. Prefix with ID: or AUTHOR: or CONTENT: to search for an article ID, article author or search in article content.

File: administrator/components/com_actionlogs/models/actionlogs.php [hunk 3]
```diff
@@ -149,6 +149,10 @@ protected function getListQuery()
 			{
 				$query->where($db->quoteName('a.item_id') . ' = ' . (int) substr($search, 8));
 			}
+			elseif (strpos($search, 'uid:') === 0)
```
Comment (Quy) [inline administrator/components/com_actionlogs/models/actionlogs.php]:
```suggestion
			elseif (stripos($search, 'uid:') === 0)
```

**Comment by pravinTek** [comment]
Hi,

I have checked & search user but never get any errors...



While applying given the patch that time getting below error:
The file marked for modification does not exist: **administrator/components/com_actionlogs/models/actionlogs.php**

<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26754">issues.joomla.org/tracker/joomla-cms/26754</a>.</sub>

File: administrator/components/com_actionlogs/models/forms/filter_actionlogs.xml [hunk 4]
```diff
@@ -4,7 +4,7 @@
 		<field
 			name="search"
 			type="text"
-			label="COM_ACTIONLOGS_FILTER_SEARCH_DESC"
+			description="COM_ACTIONLOGS_FILTER_SEARCH_DESC"
```
Comment (wilsonge) [inline administrator/components/com_actionlogs/models/forms/filter_actionlogs.xml:7]: Why are we removing the label here? Surely it’s still required for a11y?

Comment (brianteeman) [reply administrator/components/com_actionlogs/models/forms/filter_actionlogs.xml:7]: the label is already there as it comes from the layout - check the source and you will see

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on ef85fd5af10df906f983f8493f54a205fd3e9c97<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26754">issues.joomla.org/tracker/joomla-cms/26754</a>.</sub>

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on ef85fd5af10df906f983f8493f54a205fd3e9c97<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26754">issues.joomla.org/tracker/joomla-cms/26754</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26754">issues.joomla.org/tracker/joomla-cms/26754</a>.</sub>

**Comment by HLeithner** [comment]
please remove the uid feature, thx

**Comment by brianteeman** [comment]
uid removed as requested

**Comment by HLeithner** [comment]
Thanks

**Comment by brianteeman** [comment]
Thanks

**Pull request merged**